prow/flagutil/github.go

```diff
@@ -31,6 +31,7 @@ import (
 // GitHubOptions holds options for interacting with GitHub.
 type GitHubOptions struct {
 	endpoint            Strings
+	gitEndpoint         string
```

Comment:
Is it valid for `gitEndpoint` and `endpoint` to be on two different domains?

Reply:
Yes, here people can put whatever they want.

Do you have an idea on validation or enforcement of this?